### ISSUE TITLE
Linux Entries are listed multiple times and mix up between disks

### ISSUE THREAD

**Issue by knightsamar** [open-power/petitboot#21]
version: dev.20150401 on an IBM 8247-42L machine

We have two Linux installations one from an old LPAR-based setup and one that came pre-installed with the machine.

The entries mix up and are listed multiple times with no easy way to distinguish.
```
  [Disk: sdd2 / 6cf91c84-1d44-4611-9e69-1d37ca72e3b7]
    Ubuntu 14.04.2 LTS (14.04) (on /dev/sdc2)
    Ubuntu (on /dev/sdc2)
    Ubuntu, with Linux 3.16.0-30-generic (on /dev/sdc2)
    Ubuntu, with Linux 3.16.0-30-generic (recovery mode) (on /dev/sdc2)
    Ubuntu, with Linux 3.13.0-62-generic (on /dev/sdc2)
    Ubuntu, with Linux 3.13.0-62-generic (recovery mode) (on /dev/sdc2)
    Ubuntu, with Linux 3.19.0-25-generic (recovery mode)
    Ubuntu, with Linux 3.19.0-25-generic
    Ubuntu
  [Disk: sdf2 / 40652b06-fed4-4cdb-ba65-bec98c951927]
    Ubuntu, with Linux 3.13.0-62-generic (recovery mode)
    Ubuntu, with Linux 3.13.0-62-generic
    Ubuntu, with Linux 3.16.0-30-generic (recovery mode)
    Ubuntu, with Linux 3.16.0-30-generic
    Ubuntu
```

We have to carefully check each potential entry's root UUID to figure out the right one (We have by now documented which UUID is for which installation).

Also, the disk numbers (sdf, sdh) keep changing between boots, but I guess that is a different and trivial issue altogether.

**Comment by knightsamar** [comment]
I found that multiple such entries did actually exist in the grub.cfg files produced by the GRUB installation on the LPAR. 

If petitboot does take data from the grub.cfg in each installation, then I guess this bug does not have anything to do with petitboot.

**Comment by sammj** [comment]
> I found that multiple such entries did actually exist in the grub.cfg files produced by the GRUB installation on the LPAR.

Yep, that's what I suspected was the cause here. GRUB by default will scan all available disks rather than just the disk it is on when updating its config. You can prevent this by setting
`GRUB_DISABLE_OS_PROBER=true` in `/etc/default/grub` and updating the GRUB config.

**Comment by shenki** [comment]
Would it be possible for the distros to set this by default when installing on a Power machine?

**Comment by sammj** [comment]
> Would it be possible for the distros to set this by default when installing on a Power machine?

Yep that might be a good idea - it would better fit the majority of use cases. Depends if we can convince them :)
